tt_metal/api/tt-metalium/logger.hpp

```diff
@@ -15,10 +15,29 @@
 #include <pybind11/iostream.h>
 #endif
 
+#include <fmt/format.h>
 #include <fmt/color.h>
 #include <fmt/core.h>
 #include <fmt/ostream.h>
 
+#include <spdlog/spdlog.h>
+
+// Forwarding Macros
+#define TT_LOG_INFO(fmt, ...) SPDLOG_INFO(fmt, ##__VA_ARGS__)
+#define TT_LOG_INFO_WITH_CAT(category, fmt, ...) SPDLOG_INFO("[{}] | " fmt, category, ##__VA_ARGS__)
+
+#define TT_LOG_WARN(fmt, ...) SPDLOG_WARN(fmt, ##__VA_ARGS__)
+#define TT_LOG_WARN_WITH_CAT(category, fmt, ...) SPDLOG_WARN("[{}] | " fmt, category, ##__VA_ARGS__)
+
+#define TT_LOG_ERROR(fmt, ...) SPDLOG_ERROR(fmt, ##__VA_ARGS__)
+#define TT_LOG_ERROR_WITH_CAT(category, fmt, ...) SPDLOG_ERROR("[{}] | " fmt, category, ##__VA_ARGS__)
+
+#define TT_LOG_DEBUG(fmt, ...) SPDLOG_DEBUG(fmt, ##__VA_ARGS__)
+#define TT_LOG_DEBUG_WITH_CAT(category, fmt, ...) SPDLOG_DEBUG("[{}] | " fmt, category, ##__VA_ARGS__)
+
+#define TT_LOG_TRACE(fmt, ...) SPDLOG_TRACE(fmt, ##__VA_ARGS__)
+#define TT_LOG_TRACE_WITH_CAT(category, fmt, ...) SPDLOG_TRACE("[{}] | " fmt, category, ##__VA_ARGS__)
```

Comment:
Can we just use `logger->set_pattern` instead of coding this in macro? Also why not make `category` it's own logger?